src/reader/JsonFormatter.cpp

```diff
@@ -59,6 +59,8 @@ const std::string opt_split = std::string("split");
 const std::string opt_pretty = std::string("pretty");
 const std::string opt_quote_all = std::string("quote-all");
 
+bool g_first_row = true;
```

Comment:
You need to reset this in JsonFormatter::flush() @alfredo-gimenez

Reply:
And make it an atomic while you're at it

Reply:
In fact, was the problem just a race condition? I don't think this has to be a global variable, just needs to be protected between threads.

Reply:
just saw this, yea it was a race condition. I previously thought every thread was getting it's own `m_first_row`, is that not the case then? I can change it back to class scope if so, it's now thread-safe on the latest.

Reply:
There is only one formatter instance for all threads. However, you can create multiple formatters over time if you do multiple flushes. Better put it back in the class scope.

Reply:
sounds good, done.